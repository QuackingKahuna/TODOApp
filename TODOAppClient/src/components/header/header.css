.Header-logo {
    height: 10vmin;
    pointer-events: none;
}

.header {
    background-color: white;
    min-height: 12vh;
    min-width: 100vh;
    display: flex;
    align-items: center;
    justify-content: space-between;
    font-size: calc(10px + 2vmin);
    color: #039dfc;
    border-bottom: 1px solid #039dfc;
    padding: 0vh 5vh;
    box-shadow: 0vh 0.1vh 1vh #039dfc;
}