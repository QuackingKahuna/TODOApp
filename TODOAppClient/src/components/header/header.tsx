import React from 'react';
import logo from './../../logo.svg';
import "./header.css"

export const Header: React.FC = () => {
    return (
        <div className="header">
            <span data-testid="name">Barclays TODO Assesment</span>
            <img src={logo} className="Header-logo" alt="logo" />
        </div>
    )
}