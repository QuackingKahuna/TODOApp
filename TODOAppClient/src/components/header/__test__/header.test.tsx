import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

import { Header } from "../header";

afterEach(cleanup);
it("header renders without crashing", () => {
    render(<Header/>);
})

it("header includes Barclays TODO Assesment", () => {
    render(<Header/>);
    const name = screen.getByTestId("name");
    expect(name).toHaveTextContent("Barclays TODO Assesment");
})

it("matches snapshot", () => {
    const tree = renderer.create(<Header/>).toJSON();
    expect(tree).toMatchSnapshot();
})