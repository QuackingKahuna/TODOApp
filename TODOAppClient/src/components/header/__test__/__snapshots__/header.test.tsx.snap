// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="header"
>
  <span
    data-testid="name"
  >
    Barclays TODO Assesment
  </span>
  <img
    alt="logo"
    className="Header-logo"
    src="logo.svg"
  />
</div>
`;
